vulture_os/workflow/config/haproxy_portal.conf

```diff
@@ -122,35 +120,47 @@ frontend Workflow_{{conf.id}}
 
     ### ACCESS ###
     # Workflow {{conf.name}}
+
+    # get header part of the JWT
+    http-request set-var(txn.alg) http_auth_bearer,jwt_header_query('$.alg')
+
+    # get payload part of the JWT
+    http-request set-var(txn.iss) http_auth_bearer,jwt_payload_query('$.iss')
+    http-request set-var(txn.aud) http_auth_bearer,jwt_payload_query('$.aud')
+    http-request set-var(txn.exp) http_auth_bearer,jwt_payload_query('$.exp','int')
+    http-request set-var(txn.now) date()
+
+    {% for repo in conf.authentication.openid_repos if repo.enable_jwt -%}
+    # Validate JWT for repo {{repo.name}}
+    http-request set-var(txn.valid_jwt) bool(true) if { var(txn.alg) -m str "{{repo.jwt_signature_type}}" } { var(txn.iss) -m str "{{repo.provider_url}}" } {% if repo.jwt_validate_audience -%} { var(txn.aud) -m str "{{conf.redirect_uri}}" } { http_auth_bearer,jwt_verify(txn.alg,"{{repo.jwt_key if repo.jwt_signature_type in ("HS256", "HS384", "HS512") else repo.get_jwt_key_filename()}}") -m int 1 } {%- endif %} { var(txn.exp),sub(txn.now) -m int gt 0 }
+    {% endfor -%} {# for repo in conf.authentication.openid_repos if repo.enable_jwt #}
+
     # Check session with LUA
```

Comment:
needs to be rewritten to validate all conditions at once: here conditions between different OpenID JWT parameters can be mixed